src/Processors/Formats/Impl/PrometheusTextOutputFormat.cpp

```diff
@@ -0,0 +1,341 @@
+#include <Processors/Formats/Impl/PrometheusTextOutputFormat.h>
+
+#include <optional>
+#include <type_traits>
+
+#include <base/defines.h>
+#include <base/types.h>
+
+#include <Columns/ColumnMap.h>
+#include <Columns/IColumn.h>
+
+#include <Common/assert_cast.h>
+
+#include <DataTypes/DataTypeMap.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/Serializations/ISerialization.h>
+
+#include <Formats/FormatFactory.h>
+#include <Formats/registerWithNamesAndTypes.h>
+
+#include <IO/ReadBufferFromString.h>
+#include <IO/readFloatText.h>
+#include <IO/WriteBufferFromString.h>
+#include <IO/WriteHelpers.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+constexpr auto FORMAT_NAME = "Prometheus";
+
+static bool isDataTypeString(const DataTypePtr & type)
+{
+    return WhichDataType(type).isStringOrFixedString();
+}
```

Comment:
We already have such function:
https://github.com/ClickHouse/ClickHouse/blob/a1e54c39185298a4cc3a14d3381f10ab8871db54/src/DataTypes/IDataType.h#L504-L508